File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-6.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = 0
        l, r = 0, 0
        replacements = 0
        char_map = {}
        while r < len(s):
            if s[r] not in char_map:
                char_map[s[r]] = 1
            else:
                char_map[s[r]] += 1
            for v in char_map.values():
                replacements = max(replacements, v)
            while (r -l + 1) - replacements > k:
                char_map[s[l]] -= 1
                l += 1
            res = max(res, r - l + 1)
            r += 1
        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for(int i : nums){
            set.add(i);
        }
        int res = 0;

        for(int i : set){
            if(!set.contains(i-1)){
                int len = 1;
                while(set.contains(i+len)){
                    len++;
                }
                res = Math.max(res, len);
            }
        }
        return res;
    }
}
